updater.js

```diff
@@ -0,0 +1,225 @@
+  }
+
+  var parsedLockfileLines = lockfileRaw.split('\n')
+  var countLines = parsedLockfileLines.length
+  if (packageManager === 'yarn') {
+    var foundCaniuse = false
+    for (var i = 0; i < countLines; i++) {
+      var row = parsedLockfileLines[i]
+      if (row.indexOf(PACKAGE_CANIUSE) === 0) {
+        foundCaniuse = true
+        continue
+      }
+
+      if (foundCaniuse && /version/.test(row)) {
+        var rowVersion = /version "([^"]+)"/.exec(row)
+        if (rowVersion.length > 1) {
+          return rowVersion[1]
+        }
+        return null
+      }
+    }
+  } else if (packageManager === 'pnpm') {
+    for (var j = 0; j < countLines; j++) {
+      var line = parsedLockfileLines[j]
+      if (line.indexOf('/' + PACKAGE_CANIUSE) >= 0) {
+        var rule = /\/([\d.]+):/.exec(line)
+        if (rule.length > 1) {
+          return rule[1]
+        }
+        return null
+      }
+    }
+  }
+
+  return null
+}
+
+function getLastVersionInfo () {
+  try {
+    var raw = childProcess
+      .execSync('npm show ' + PACKAGE_CANIUSE + ' --json')
+      .toString()
+
+    return JSON.parse(raw)
+  } catch (e) {
+    throw new BrowserslistError(
+      'An error occurred getting information about the latest version of ' +
+      PACKAGE_CANIUSE +
+      '.\n' +
+      'Check your Internet connection.'
+    )
+  }
+}
+
+function updateLockfile (lockfileRaw, info, packageManager) {
+  var parsedLockfileLines = lockfileRaw.split('\n')
+  var countLines = parsedLockfileLines.length
+  var hasNecessaryDeps = false
+
+  if (packageManager === 'npm') {
+    for (var i = 0; i < parsedLockfileLines.length; i++) {
+      if (
+        !hasNecessaryDeps &&
+        parsedLockfileLines[i].indexOf(PACKAGE_CANIUSE) > 0 &&
+        /:\s*{/.test(parsedLockfileLines[i])
+      ) {
+        hasNecessaryDeps = true
+      }
+
+      if (!hasNecessaryDeps) {
+        continue
+      }
+
+      if (parsedLockfileLines[i].indexOf('"version"') > 0) {
+        parsedLockfileLines[i] = parsedLockfileLines[i]
+          .replace(/:\s*"([^"]+)"/, ': "' + info.version + '"')
+      } else if (parsedLockfileLines[i].indexOf('"resolved"') > 0) {
+        parsedLockfileLines[i] = parsedLockfileLines[i]
+          .replace(/:\s*"([^"]+)"/, ': "' + info.dist.tarball + '"')
+      } else if (parsedLockfileLines[i].indexOf('"integrity"') > 0) {
+        parsedLockfileLines[i] = parsedLockfileLines[i]
+          .replace(/:\s*"([^"]+)"/, ': "' + info.dist.integrity + '"')
+      } else if (parsedLockfileLines[i].indexOf('}') > 0) {
+        hasNecessaryDeps = false
+      }
+    }
+  } else if (packageManager === 'yarn') {
+    for (var j = 0; j < countLines; j++) {
+      if (
+        parsedLockfileLines[j].indexOf(PACKAGE_CANIUSE) >= 0 &&
+        parsedLockfileLines[j].indexOf('@') > 0
+      ) {
+        hasNecessaryDeps = true
+      }
+
+      if (!hasNecessaryDeps) {
+        continue
+      }
+
+      if (parsedLockfileLines[j].indexOf('version') > 0) {
```

Comment:
I suggest you use `j++` to go to next line instead of goind to next step by `for` with some state-variable.

```
for (var i = 0; i < lines.legnth; i++) {
  if (test(line[i])) {
    line[i + 1] = fix2
    line[i + 2] = fix2
    i += 2
  }
}
```